File: Java/Prj05_GeometriaPiana/src/figure/Quadrato.java
package figure;

public class Quadrato extends Rettangolo{

    private Segmento lato;

    public Quadrato(Segmento lato){
        super(lato,lato);
        this.lato = lato;
    }

    @Override
    public double perimetro() {
        return lato.lunghezza() * 4;
    }

    @Override
    public double area() {
        return Math.pow(lato.lunghezza(),2);
    }

    @Override
    public String toString() {
        return "Quadrato[" +
                "lato=" + lato +
                "Perimetro=" + perimetro() +
                ", Area=" + area() +
                ']';
    }
}

File: DotNet/ClasseQuadrato/ClasseQuadrato/Quadrato.cs
using System.Security.Cryptography.X509Certificates;

namespace ClasseQuadrato
{
	internal class Quadrato
	{
		// caratteristiche
		public double lato;



		// azioni - funzionalità
		public double Perimetro()
		{
			return lato * 4;
		}

		public double Area()
		{
			return Math.Pow(lato, 2);
		}

		public double Diagonale()
		{
			return lato * Math.Sqrt(2);
		}

	}
}
